bob-backend/src/pearl/disk_controller.rs

```diff
@@ -712,4 +712,28 @@ impl DiskController {
             .fold(0, |acc, x| async move { acc + x })
             .await
     }
+
+    async fn get_alien_results<R, RF, F, Res>(&self, op: &Operation, f: F, 
+                                              init: Res, fold: impl Fn(Res, R) -> Res) -> Option<Res>
+        where
+            RF: Future<Output = Result<R, Error>>,
+            F: Fn(Group) -> RF 
+    {
+        let groups = self.find_all_groups(op).await;
+        if groups.is_empty() {
+            return None;
+        }
+        let mut result = init;
+        for g in groups {
+            let r = f(g).await;
+            match r {
+                Ok(r) => result = fold(result, r),
+                Err(e) => {
+                    trace!("error getting alien results for op {:?}: {:?}", op, e);
+                    self.process_error(e).await;
```

Comment:
Behavior changed here. Previously only `put` and `delete` called `process_error`. Now it is called for read operations too. Looks like it should be ok here, but in backport to v2.0 later the original behavior should be preserved as much as possible